File: anagram.py
str = raw_input("Enter a string: ")
substr = raw_input ("Enter the anagram/substring to look for:")

len = len(substr);

count=0;
found=False;
for i in range(count,len+1):
	if(str[count:count+len]==substr):
		found=True
	else:
		count+=1
		found=False

if (found):
	print("Substring found")
else:
  print("substring doesn't exist")

File: anagram.js
var readline = require("readline");

var rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

var string = "";
var substr = "";

rl.question("Enter a String: ", function(answer) {
	string = answer;
	rl.question("Enter substring: ", function(answer) {
		substr = answer;
		rl.pause();
	});
});

rl.on("pause", function() {
	var len = substr.length;
	var start = 0;
	var end = len;
	var found = false;
	for (; start < string.length; ) {
		if (string.substring(start, end) == substr) {
			found = true;
			break;
		} else {
			start++;
			end++;
		}
	}

	var result = found ? "substring found" : "substring doesn't exist";
	console.log(result);
});
